sdk/tests/deqp/framework/referencerenderer/rrFragmentOperations.js

```diff
@@ -197,16 +212,14 @@ rrFragmentOperations.executeDepthCompare = function(inputFragments, depthFunc, d
                 var depthBufferValue = depthBuffer.getPixDepth(fragSampleNdx, frag.pixelCoord[0], frag.pixelCoord[1]);
                 var sampleDepthFloat = frag.sampleDepths[fragSampleNdx];
 
-                /* convert input float to target buffer format for comparison */
-                var access = new tcuTexture.PixelBufferAccess({
-                    format: depthBuffer.getFormat(),
-                    width: 1,
-                    height: 1,
-                    depth: 1,
-                    data: new ArrayBuffer(8)
-                });
-                access.setPixDepth(sampleDepthFloat, 0, 0, 0);
-                var sampleDepth = access.getPixDepth(0, 0, 0);
+                var sampleDepth;
+                if (convertToDepthBuffer) {
+                    /* convert input float to target buffer format for comparison */
+                    access.setPixDepth(sampleDepthFloat, 0, 0, 0);
+                    sampleDepth = access.getPixDepth(0, 0, 0);
+                } else {
+                    sampleDepth = deMath.clamp(sampleDepthFloat, 0.0, 1.0);
+                }
```

Comment:
Looking a little more closely at the C++ code:
https://android.googlesource.com/platform/external/deqp/+/deqp-dev/framework/referencerenderer/rrFragmentOperations.cpp#195

it looks like the floating-point sample depth is always stored using the PixelBufferAccess, and then it's fetched as an unsigned int for the depth comparison. With the current code structure in the JavaScript port, does it make sense to try to mimic this structure? It is technically feasible to reinterpret the bits of a floating-point number as an integer by storing into a Float32Array and reading back from the same ArrayBuffer via Uint32Array.